File: springapp/src/main/java/com/examly/springapp/service/EmployeeRepairMappingService.java
package com.examly.springapp.service;
import com.examly.springapp.model.EmployeeRepairMapping;
import com.examly.springapp.repository.EmployeeRepairMappingRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;


@Service
public class EmployeeRepairMappingService {
    @Autowired
    private EmployeeRepairMappingRepository employeeRepairMappingRepository;
    public List<EmployeeRepairMapping> getAllEmployeeMappings() {
        return employeeRepairMappingRepository.findAll();
    }
    public EmployeeRepairMapping assignRepairIdToEmployee(Long employeeId, EmployeeRepairMapping em) {
        EmployeeRepairMapping mapping = new EmployeeRepairMapping();
        mapping.setEmployeeId(employeeId);
        mapping.setRepairId(em.getRepairId());
        return employeeRepairMappingRepository.save(mapping);
    }
}

File: reactapp/src/services/EmployeeRepairMappingService.js
import axios from 'axios';
const API_BASE_URL = 'https://8080-eafabcbdbecabfcdcbbdbeaabbeaeaadbdbabf.project.examly.io';
const EmployeeRepairMappingService = {
  getAllEmployeeMappings: () => {
    return axios.get(API_BASE_URL+'/api/employeeRepairMapping');
  },
  assignRepairIdToEmployee: (employeeId, repairId) => {
    return axios.put( API_BASE_URL+'/api/employeeRepairMapping/assignRepairIdToEmployee/'+employeeId, repairId);
  },
  
};
export default EmployeeRepairMappingService;